apps/api/src/app/subscribers/usecases/delete-subscriber-credentials/delete-subscriber-credentials.usecase.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@nestjs/common';
+import { SubscriberRepository, IntegrationRepository, SubscriberEntity } from '@novu/dal';
+import { AnalyticsService, buildSubscriberKey, InvalidateCacheService } from '@novu/application-generic';
+
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { DeleteSubscriberCredentialsCommand } from './delete-subscriber-credentials.command';
+
+@Injectable()
+export class DeleteSubscriberCredentials {
+  constructor(
+    private invalidateCache: InvalidateCacheService,
+    private subscriberRepository: SubscriberRepository,
+    private integrationRepository: IntegrationRepository,
+    private analyticsService: AnalyticsService
+  ) {}
+
+  async execute(command: DeleteSubscriberCredentialsCommand) {
+    const foundSubscriber = await this.subscriberRepository.findBySubscriberId(
+      command.environmentId,
+      command.subscriberId
+    );
+
+    if (!foundSubscriber) {
+      throw new ApiException(`SubscriberId: ${command.subscriberId} not found`);
+    }
```

Comment:
If we replace this bit of the code with the GetSubscriber use case we benefit of 2 things:
- not duplicating code and in case of modification to be reused here
- take advantage of the read cache of the subscriber